### DISCUSSION TITLE
Guessing Game

### DISCUSSION THREAD

**Discussion by F8bi8n** [cpp-for-yourself/lectures-and-homeworks#24] [Homeworks Q&A] [answered]
I can't find a difference between the output generated by my program and the expected one:

<img width="824" alt="image" src="https://user-images.githubusercontent.com/63605568/191927305-4794a1e1-8a20-470e-8c41-764f6f7d1269.png">

Are you able to spot the mistake I made?

**Comment by niosus** [comment, marked as answer]
Hmm, sorry for a non-helpful way of showing the output difference, I have an issue for me to fix it soon: https://github.com/cpp-for-yourself/homework_checker/issues/17

As for your issue, the only issue I see is the training `'` symbol. Other than that it seems the same :thinking: 

```diff
'Welcome to the GUESSING GAME!
I will generate a number and you will guess it!
Please provide the smallest number: 
Please provide the largest number: 
I've generated a number. Try to guess it!
- Please provide the next guess: You've done it! You guessed the number 1 in 1 guesses!''
+ Please provide the next guess: You've done it! You guessed the number 1 in 1 guesses!'
```

**Reply by F8bi8n** [reply]
Thank you very much, that fixed the issue!

**Reply by niosus** [reply]
Awesome, glad I could help!

**Comment by niosus** [comment]
@FabianKuon for completeness, I've changed the way we render the output mismatch now (see https://github.com/cpp-for-yourself/homework_checker/pull/18 for details). It should use the `diff` syntax now and the differences between what the user provided and what the bot expects should be much more obvious now. 😉 

For an example of how it looks, check out the test homework in the `homework_checker` repo: https://github.com/cpp-for-yourself/homework_checker/wiki#encountered-errors
